### PR TITLE
PCHR-3008: Implement AngularJS Style Guide and JS Semi-Standard across Job Contract app

### PR DESCRIPTION
**Pull request by AkA84**
## Overview
This PR makes the Job Contract angular app and its file to follow both the AngularJS Style Guide and the JS Semi-Standard Style

## Technical Details
Please refer to the technical details in #2066, as this PR does basically the same thing

## Comment
I began to migrate `$scope` to `vm`, but then I realized that so many things could go wrong with so many controllers, and the PR was already pretty huge. So I'll leave that to the dev who is going to work on those controllers next time

----

* [x] Tests pass

### CODE REVIEW

File: hrjobcontract/js/src/job-contract/controllers/contract-list.controller.js [hunk 1]
```diff
@@ -58,75 +53,108 @@ define([
...
       for (entityName in entityServices) {
-        promiseModel[entityName] = entityServices[entityName].model(fields[entityName]);
+        promiseFields[entityName] = entityServices[entityName].getFields();
       }
 
-      return $q.all(promiseModel);
-    }).then(function (model) {
-      $scope.model = model;
+      $q.all(promiseFields).then(function (fields) {
+        $scope.fields = fields;
```
Comment (reneolivo) [inline hrjobcontract/js/src/job-contract/controllers/contract-list.controller.js]: Can you please explain in the PR description why this was renamed from `models` to `fields`?

File: hrjobcontract/js/karma.conf.js [hunk 2]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (reneolivo) [inline hrjobcontract/js/karma.conf.js]: why is there an empty line here?

File: hrjobcontract/gulpfile.js [hunk 3]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (reneolivo) [inline hrjobcontract/gulpfile.js]: why is there an empty line here?

File: hrjobcontract/js/src/job-contract/controllers/contract.controller.js [hunk 4]
```diff
@@ -110,9 +110,8 @@ define([
         _.extend(entity.health, revisionDetails.health);
         _.extend(entity.pay, revisionDetails.pay);
         _.extend(entity.pension, revisionDetails.pension);
-        _.each(entity.leave, function (leaveType, leaveTypeId) {
-          _.extend(leaveType, revisionDetails.leave ? revisionDetails.leave[leaveTypeId] : '');
-        });
+
```
Comment (igorpavlov) [inline hrjobcontract/js/src/job-contract/controllers/contract.controller.js]: This empty line is not needed I think

File: hrjobcontract/js/src/job-contract/controllers/contract.controller.js [hunk 5]
```diff
@@ -370,9 +387,7 @@ define([
       _.extend($scope.health, newScope.health || contractRevisionIdObj);
       _.extend($scope.pension, newScope.pension || contractRevisionIdObj);
```
Comment (igorpavlov) [inline hrjobcontract/js/src/job-contract/controllers/contract.controller.js]: This empty line is not needed I think

**Pull request merged**